client/dashboard/me/security-two-step-auth/hooks/use-register-security-key.ts

```diff
@@ -0,0 +1,64 @@
+import {
+	securityKeyRegistrationChallengeQuery,
+	validateSecurityKeyRegistrationMutation,
+} from '@automattic/api-queries';
+import { create, supported } from '@github/webauthn-json';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { __ } from '@wordpress/i18n';
+
+function isBrowser() {
+	try {
+		if ( ! window ) {
+			return false;
+		}
+	} catch ( err ) {
+		return false;
+	}
+	return true;
+}
+
+export const isWebAuthnSupported = () => {
+	return isBrowser() && supported();
+};
+
+export const useRegisterSecurityKey = () => {
+	const queryClient = useQueryClient();
+
+	return useMutation( {
+		mutationFn: async ( keyName: string ) => {
+			try {
+				// First, get the registration challenge
+				const options = await queryClient.fetchQuery( securityKeyRegistrationChallengeQuery() );
+
+				// Create the WebAuthn credential
+				const credential = await create( { publicKey: options } );
+
+				// Validate the registration with the server
+				const validationData = {
+					data: JSON.stringify( credential ),
+					name: keyName,
+				};
+				return await validateSecurityKeyRegistrationMutation().mutationFn?.( validationData );
+			} catch ( error ) {
+				// Handle WebAuthn specific errors with user-friendly messages
+				if ( error && typeof error === 'object' && 'name' in error ) {
```

Comment:
You could use the `isWpError()` function here.

https://github.com/Automattic/wp-calypso/blob/a0bba7ebce3ef5b330b57a472181b1cb64dbde3c/packages/api-core/src/error.ts#L22

Since the `wpcom` package throws `WPError` errors.

Reply:
Hmm. This doesn't work. It looks like the error is thrown by the '@github/webauthn-json'